apps/web/components/eventtype/EventTeamTab.tsx

```diff
@@ -207,13 +276,31 @@ const ChildrenEventTypes = ({
 }: {
   childrenEventTypeOptions: ReturnType<typeof mapMemberToChildrenOption>[];
 }) => {
+  const formMethods = useFormContext<FormValues>();
+  const [assignAllTeamMembers, setAssignAllTeamMembers] = useState(
+    formMethods.getValues("assignAllTeamMembers") ?? false
+  );
+
   return (
-    <Controller<FormValues>
-      name="children"
-      render={({ field: { onChange, value } }) => (
-        <ChildrenEventTypesList value={value} options={childrenEventTypeOptions} onChange={onChange} />
-      )}
-    />
+    <div className="border-subtle mt-6 space-y-5 rounded-lg border px-4 py-6 sm:px-6">
+      <div className="flex flex-col gap-4">
+        <AssignAllTeamMembers
```

Comment:
It throws an error when 'Automatically add to all team members' gets selected and there were already children selected before
<img width="336" alt="Screenshot 2024-01-22 at 1 43 13 PM" src="https://github.com/calcom/cal.com/assets/30310907/cecef46a-5186-4b34-b84b-74f5a382dd50">
<img width="907" alt="Screenshot 2024-01-22 at 1 43 31 PM" src="https://github.com/calcom/cal.com/assets/30310907/02548d0d-0cd2-4e31-8c84-a5022a82c1bd">

Reply:
This was a weird issue, for some reason when we checked assignAllTeamMembers it was not filtering the selected eventType slug from the selected children's owner's eventTypeSlugs so have added that in the code and it's working fine.